_sources/functions_in_cpp/mixed_up_code.rst

```diff
@@ -0,0 +1,344 @@
+   :numbered: left
+   :adaptive:
+   :practice: T
+
+   Let's write the code for the isDoubleDigit function. isDoubleDigit
+   takes num as a parameter. isDoubleDigit returns true if 
+   num is a double digit number and returns false otherwise. 
+   In the code, use the if then else structure. 
+   Put the necessary blocks of code in the correct order.
+   -----
+   bool isDoubleDigit (int num) {
+   =====
+   isDoubleDigit (int num) {  #paired
+   =====
+      if (num >= 10 && num < 100) {
+   =====
+      if (10 <= num <= 99) {  #distractor
+   =====
+      if (num > 10 && num < 100) {  #distractor
+   =====
+      if (num > 10 && num <= 100) {  #distractor
+   =====
+         return true;
+   =====
+      }
+   =====
+      else {
+   =====
+         return false;
+   =====
+      }
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_8
+   :numbered: left
+   :adaptive:
+   :practice: T
+
+   Let's write the code for the Compare function. Compare
+   takes two integers a, b. Compare returns 1 if 
+   a is greater than b, -1 if a is less than b and 0 if they are equal.
+   Put the necessary blocks of code in the correct order.
+   -----
+   int Compare (int a, int b) {
+   =====
+   bool Compare (int a, int b) {   #paired
+   =====
+      if (a > b) {
+   =====
+      if (a > b && a &lt b) {  #distractor
+   =====
+         return 1;
+   =====
+      }
+   =====
+      else if (a &lt b) {  
+   =====
+         return -1;
+   =====
+      }
+   =====
+      else if (a > 0){ #distractor
+   =====
+      else {
+   =====
+         return 0;
+   =====
+      }
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_9
+   :numbered: left
+   :adaptive:
+   :practice: T
+
+   Let's write the code for the isFactor function. isFactor
+   takes two parameters, num and factor.
+   isFactor returns true if factor is a factor of num 
+   and returns false otherwise. In the code, use the if then 
+   else structure. Put the necessary blocks of code in the 
+   correct order.
+   -----
+   bool isFactor (int num, int factor) {
+   =====
+   void isFactor (int num, int factor) {  #paired
+   =====
+      if (num % factor == 0) {
+   =====
+      if (num / factor == 0) {  #distractor
+   =====
+      if (num % factor) {  #distractor
+   =====
+      if (factor % num == 0) {  #distractor
+   =====
+         return true;
+   =====
+      }
+   =====
+      else {
+   =====
+         return false;
+   =====
+      }
+   =====
+   }
+
+.. parsonsprob:: functions_in_cpp_mixed_up_code_10
+   :numbered: left
+   :adaptive:
+   :practice: T
+
+   Let's write the code for the isPerfectSquare function. isPerfectSquare
+   takes input as a parameter and returns true if input is a 
+   perfect square and returns false otherwise. A perfect square is a number
+   that has an integer square root. In the code, use the if then
+   else structure. Put the necessary blocks of code in the correct order.
```

Comment:
I think there needs to be a little more explanation about how to tell if a number is a perfect square. Also, I have the same comment about this function as I had about the isDoubleDigit function.